docs/getting-started/prerequisites.md

```diff
@@ -85,21 +98,67 @@ Our supported systems include:
 - Note: Only x86 architectures are currently supported
 
 To install the dependencies, run the script:
+
 ```bash
 local/install_deps.bash
 ```
 
 [homebrew]: https://brew.sh/
 
 ## Loading pipenv
+
 After you run the `local/install_deps.bash` script, activate pipenv by running the following command:
 
 ```bash
 pipenv shell
 ```
+
 This loads all the Python dependencies in the current environment.
 
 You can verify that everything works by running:
+
 ```bash
 python butler.py --help
 ```
+
+### Debugging Common Dependency Issues
+
+If you are having trouble installing dependencies due to Python versioning, try:
+
+```
+$ PYTHON=python3.7 ./local/install_deps.bash
+```
+
+Then run:
+
+```
+$ pipenv shell
+$ python --version
+```
+
+The version should be the one you built from source and used to build your dependencies. If it is not, set the pipenv shell version of Python:
+
+```
+$ pipenv install --python 3.7
+```
+
+Assuming your shell has the expected version of Python assigned, the following command should run all appengine tests:
+
+```
+python3.7 butler.py py_unittest -t appengine
+```
+
+If, at this point, you see an error to the effect that the config.yaml is missing, sync the pipenv environment’s dependencies with the outer environment’s dependencies:
+
+```
+$ cd src; pipenv sync
+```
+
+#### Refreshing the Python Environment
+
+You might get stuck in a mode where the python version is the global value, rather than the version you set for the pipenv environment - even when apparently leaving the virtual environment through the `exit`command. If this happens, you may still be in a virtual environment (especially if you see `(clusterfuzz)`before the command prompt). To escape this virtual environment, use the command `deactivate`, then run `python3.7 -m pipenv shell` again to start the pipenv shell with the correct python version.
+
+## ClusterFuzz Development Tips
+
+- Before committing, run `python butler.py format` to clean up formatting
+- Make sure you lint all updated files. Either with `python butler.py lint` or individually with `pylint --score=no --jobs=0 <name of directory>`
```

Comment:
Better to do `python butler.py lint` since linting all files is necessary when you change something like a function used in another module.

Reply:
Agreed that this is the preferred way. However, my experience was that I ran into problems running `python butler.py lint`in my local environment, and so it was useful to know a workaround.

I updated to make it clearer what the preferred strategy is here. If you'd prefer that I leave out this alternative entirely, I'll go ahead and remove it - but it would have been useful to me when I was getting started!